test/e2e/kubectl/logs.go

```diff
@@ -209,4 +257,100 @@ var _ = SIGDescribe("Kubectl logs", func() {
+				if err != nil {
+					framework.Failf("Failed to get pods for Deployment: %v", err)
+				}
+
+				podOne := pods.Items[0].GetName()
+				podTwo := pods.Items[1].GetName()
+
+				ginkgo.By("expecting logs from both replicas in Deployment")
+				out := e2ekubectl.RunKubectlOrDie(ns, "logs", fmt.Sprintf("deploy/%s", deployName), "--all-pods")
+				framework.Logf("got output %q", out)
+				logLines := strings.Split(out, "\n")
+				logFound := false
+				for _, line := range logLines {
+					var deployPod bool
+					if line != "" {
+						if strings.Contains(line, fmt.Sprintf("[pod/%s/container-2]", podOne)) || strings.Contains(line, fmt.Sprintf("[pod/%s/container-2]", podTwo)) {
+							logFound = true
+							deployPod = true
+						}
+						gomega.Expect(deployPod).To(gomega.BeTrueBecause("each log should be from the default container from each pod in the Deployment"))
+					}
+
+				}
+				gomega.Expect(logFound).To(gomega.BeTrueBecause("log should be present"))
+			})
+
+			ginkgo.It("should get logs from each pod and each container in Deployment", func(ctx context.Context) {
+				ginkgo.By("Waiting for Deployment pods to be running.")
+
+				pods, err := e2edeployment.GetPodsForDeployment(ctx, c, deploy)
+				if err != nil {
+					framework.Failf("Failed to get pods for Deployment: %v", err)
+				}
+
+				podOne := pods.Items[0].GetName()
+				podTwo := pods.Items[1].GetName()
+
+				ginkgo.By("all containers and all containers")
+				out := e2ekubectl.RunKubectlOrDie(ns, "logs", fmt.Sprintf("deploy/%s", deployName), "--all-pods", "--all-containers")
+				framework.Logf("got output %q", out)
+				logLines := strings.Split(out, "\n")
+				logFound := false
+				for _, line := range logLines {
+					if line != "" {
+						var deployPodContainer bool
+						if strings.Contains(line, fmt.Sprintf("[pod/%s/container-1]", podOne)) || strings.Contains(line, fmt.Sprintf("[pod/%s/container-1]", podTwo)) || strings.Contains(line, fmt.Sprintf("[pod/%s/container-2]", podOne)) || strings.Contains(line, fmt.Sprintf("[pod/%s/container-2]", podTwo)) {
+							logFound = true
+							deployPodContainer = true
+						}
+						gomega.Expect(deployPodContainer).To(gomega.BeTrueBecause("each log should be from all containers from all pods in the Deployment"))
+					}
+				}
+				gomega.Expect(logFound).To(gomega.BeTrueBecause("log should be present"))
+				gomega.Expect(strings.Contains(out, fmt.Sprintf("[pod/%s/container-1]", podOne))).To(gomega.BeTrueBecause("pod 1 container 1 log should be present"))
+				gomega.Expect(strings.Contains(out, fmt.Sprintf("[pod/%s/container-2]", podOne))).To(gomega.BeTrueBecause("pod 1 container 2 log should be present"))
+				gomega.Expect(strings.Contains(out, fmt.Sprintf("[pod/%s/container-1]", podTwo))).To(gomega.BeTrueBecause("pod 1 container 1 log should be present"))
+				gomega.Expect(strings.Contains(out, fmt.Sprintf("[pod/%s/container-2]", podTwo))).To(gomega.BeTrueBecause("pod 1 container 2 log should be present"))
```

Comment:
I think this should say `pod 2`.

Reply:
haha yes, great eye, and attention to detail. Thanks!